python/ray/_private/node.py

```diff
@@ -83,6 +83,7 @@ def __init__(
         )
         self.all_processes: dict = {}
         self.removal_lock = threading.Lock()
+        self.cluster_id = ray_params.cluster_id
```

Comment:
I'd suggest to remove this line and only set it inside `_init_gcs_client`. This way we make sure `self.cluster_id` can never be None.